### ISSUE TITLE
static code analysis revealed some issues

### ISSUE THREAD

**Issue by kt97679** [yrutschle/sslh#190]
I performed static code analysis of sslh using pvs-studio (https://www.viva64.com/en/pvs-studio/). I did following on the ubuntu 14.04 machine:
```
wget -q -O - http://files.viva64.com/etc/pubkey.txt |   sudo apt-key add -
sudo wget -O /etc/apt/sources.list.d/viva64.list   http://files.viva64.com/etc/viva64.list
sudo apt-get update
sudo apt-get install pvs-studio
cd /tmp
git clone https://github.com/yrutschle/sslh.git
cd sslh
for x in *.c *.h; do cp $x /tmp; echo -e "// This is an open source non-commercial project. Dear PVS-Studio, please check it.\n// PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com" >$x; cat /tmp/$x >>$x; done
sudo apt-get install libconfig8-dev libpcre3-dev
pvs-studio-analyzer trace -- make
pvs-studio-analyzer analyze -o /tmp/sslh.log
plog-converter -a GA:1,2 -t tasklist -o /tmp/sslh.report /tmp/sslh.log
```
the content of /tmp/sslh.report is:
```
www.viva64.com/en/w     1       err     Help: The documentation for all analyzer warnings is available here: https://www.viva64.com/en/w/.
/tmp/sslh/common.c        134     warn    V547 Expression '!addr' is always false.
/tmp/sslh/common.c        688     warn    V547 Expression 'fn == - 1' is always false.
/tmp/sslh/sslh-select.c   431     err     V535 The variable 'j' is being used for this loop and for the outer loop. Check lines: 385, 431.
```
Please note, that since I had to add 2 lines at the beginning of each file you will need to subtract 2 from line numbers when viewing in your git repository.

**Comment by yrutschle** [comment]
Thanks for this; I have a long-term fantasy of running at
least probe.c under ASTREE, but I can't seem to get my hands
on it.

134: It's nice to see the analyser agree with me :) The test
is a sanity robustness test which dies if it succeeds as it
makes no sense (there are a few others like this,
actually). I don't plan to remove it.

688: is a real (fairly harmless) bug, fixed now in a1cc399ae57dd5487a7a064a83ad9cde172fb8df

431: The analyser 'missed' the fact that the loop is
executed in a child process which exits afterwards, so
there is no issue here. I don't think I'll change it for
now. Bizarrely it missed the fact that we also re-use i, which
is also used as a top-level loop index.

**Comment by kt97679** [comment]
Thanks for the clarifications and fix @yrutschle !
